sdk/identity/identity/review/identity.api.md

```diff
@@ -431,12 +431,24 @@ export class WorkloadIdentityCredential implements TokenCredential {
 }
 
 // @public
-export interface WorkloadIdentityCredentialOptions extends MultiTenantTokenCredentialOptions, AuthorityValidationOptions {
+export interface WorkloadIdentityCredentialKubernetesOptions extends MultiTenantTokenCredentialOptions, AuthorityValidationOptions {
     clientId?: string;
+    serviceConnectionId?: never;
     tenantId?: string;
     tokenFilePath?: string;
 }
 
+// @public
+export type WorkloadIdentityCredentialOptions = WorkloadIdentityCredentialKubernetesOptions | WorkloadIdentityCredentialServiceConnectionOptions;
```

Comment:
What is the difference between WorkloadIdentityCredentialKubernetesOptions and WorkloadIdentityCredentialServiceConnectionOptions ? They look identical

Reply:
One has federatedTokenFilePath required and the other has serviceConnectionId required

Reply:
Oh I see! I missed the `never` on each side.

This makes more sense now, but I wonder if `undefined` is a better type than `never` here - because the union of `<type>` and `never` is just `<type>` this ends up being `undefined` anyway. 

What I mean is, try this in a [typescript playground](https://www.typescriptlang.org/play?#code/C4TwDgpgBAYg9nKBeKBnYAnAlgOwOZQA+UOEAbhBkA):

```ts
type Foo = string | never
```

If you hover over `Foo` in the editor you'll notice that the type is just `string` - TypeScript is smart enough to know that and never can not be in the set of values here. When a user tries to set both, the error message will say:

```
Type 'string' is not assignable to type 'undefined'.ts(2322)
```

Which can be confusing.

So what I am suggesting is to change `never` to explicit `undefined` as a starting point. It'll make for less confusing error messages. Does that make sense? Happy to discuss more or brainstorm an API shape that provides better devex!

Reply:
Tagging @bterlson to discuss which type should we use here for  forbidden fields.

Reply:
Thanks Karishma - happy to learn one way or the other 👍